src/frontend/components/Felleskomponenter/JaNeiSpm/JaNeiSpmForSanity.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { ReactNode, useEffect, useState } from 'react';
+
+import { v4 as uuidv4 } from 'uuid';
+
+import { Box } from '@navikt/ds-react';
+import { ESvar, JaNeiSpørsmål } from '@navikt/familie-form-elements';
+import { Felt, ISkjema } from '@navikt/familie-skjema';
+
+import { useApp } from '../../../context/AppContext';
+import { AlternativtSvarForInput } from '../../../typer/common';
+import { FlettefeltVerdier, ISanitySpørsmålDokument } from '../../../typer/sanity/sanity';
+import { SkjemaFeltTyper } from '../../../typer/skjema';
+import { logSpørsmålBesvart } from '../../../utils/amplitude';
+import TekstBlock from '../Sanity/TekstBlock';
+
+interface IJaNeiSpmProps {
+    skjema: ISkjema<SkjemaFeltTyper, string>;
+    felt: Felt<ESvar | null>;
+    tilleggsinfo?: ReactNode;
+    inkluderVetIkke?: boolean;
+    spørsmålDokument: ISanitySpørsmålDokument;
+    flettefelter?: FlettefeltVerdier;
+}
+
+const JaNeiSpm: React.FC<IJaNeiSpmProps> = ({
+    skjema,
+    felt,
+    tilleggsinfo,
+    inkluderVetIkke = false,
+    spørsmålDokument,
+    flettefelter,
+}) => {
+    const [mounted, settMounted] = useState(false);
+    const { søknad, tekster, plainTekst } = useApp();
+    const { ja, nei, jegVetIkke } = tekster().FELLES.frittståendeOrd;
+
+    useEffect(() => {
+        if (mounted) {
+            spørsmålDokument &&
+                logSpørsmålBesvart(
+                    spørsmålDokument.api_navn,
+                    felt.verdi ?? AlternativtSvarForInput.UKJENT,
+                    søknad.søknadstype
+                );
+        }
+        settMounted(true);
+    }, [felt.verdi]);
+
+    return felt.erSynlig ? (
```

Comment:
Her kan du skrive 

```
if (!felt.erSynlig) {
    return null
}

return (
    <div> ... </div>
)
```

Reply:
Jeg tror det er best å la den være som den er ettersom samme logikk brukes i andre komponenter. Dersom vi blander ulike måter å håndtere det på er jeg redd det kan bli vanskelig å forstå for andre som skal lese koden.
![image](https://github.com/user-attachments/assets/c87a5c22-970d-4ec2-bdcf-789b8809b2ec)

Eventuelt kunne det lages en egen oppgave for å fikse dette i både BA og KS søknad slik denne typen logikk er konsistent overalt.

Reply:
Syns ikke det nødvendigvis er noe galt å gradvis gå over til et annet pattern. Begge måtene er jo vanlig i react også. Personlig syns jeg det er lettere å lese på den måten jeg foreslå, men om du foretrekker å beholde det på den gamle måten er det også innenfor.  Men jeg ser absolutt poenget ditt med at det er lurt å være konsekvent. 😊

Reply:
Jeg lar den stå slik som den er for å gjøre logikken konsekvent på tvers av komponenter slik at det forhåpentligvis blir lettere for andre utviklere å sette seg inn i koden.